src/overlays/actors/ovl_Fishing/z_fishing.c

```diff
@@ -3886,46 +3901,47 @@ void Fishing_UpdateFish(Actor* thisx, PlayState* play2) {
                            6.0f);
 
-            D_80B7E144 = 188.0f;
+            sRodLineSpooled = 188.0f;
 
             if (this->unk_17A[0] <= 50) {
-                switch (this->unk_1D5) {
+                switch (this->keepState) {
                     case 0:
                         if ((Message_GetState(&play->msgCtx) == TEXT_STATE_CHOICE) ||
                             (Message_GetState(&play->msgCtx) == TEXT_STATE_NONE)) {
                             if (Message_ShouldAdvance(play)) {
                                 Message_CloseTextbox(play);
                                 if (play->msgCtx.choiceIndex == 0) {
-                                    if (D_80B7A670 == 0.0f) {
-                                        D_80B7A670 = this->unk_1AC;
-                                        D_80B7E07C = this->unk_150;
-                                        D_80B7E07E = D_80B7E0B6;
+                                    if (sFishOnHandLength == 0.0f) {
+                                        sFishOnHandLength = this->fishLength;
+                                        sFishOnHandisLoach = this->isLoach;
+                                        sLureCaughtWith = sLureEquipped;
                                         Actor_Kill(&this->actor);
-                                    } else if ((this->unk_150 == 0) && (D_80B7E07C == 0) &&
-                                               ((s16)this->unk_1AC < (s16)D_80B7A670)) {
-                                        this->unk_1D5 = 1;
+                                    } else if ((this->isLoach == 0) && (sFishOnHandisLoach == 0) &&
+                                               ((s16)this->fishLength < (s16)sFishOnHandLength)) {
+                                        this->keepState = 1;
                                         this->unk_17A[0] = 0x3C;
+
                                         Message_StartTextbox(play, 0x4098, NULL);
                                     } else {
-                                        f32 temp1 = D_80B7A670;
-                                        s16 temp2 = D_80B7E07C;
-                                        D_80B7A670 = this->unk_1AC;
-                                        D_80B7E07C = this->unk_150;
-                                        D_80B7E07E = D_80B7E0B6;
-                                        this->unk_1AC = temp1;
-                                        this->unk_150 = temp2;
+                                        f32 temp1 = sFishOnHandLength;
+                                        s16 temp2 = sFishOnHandisLoach;
+                                        sFishOnHandLength = this->fishLength;
+                                        sFishOnHandisLoach = this->isLoach;
+                                        sLureCaughtWith = sLureEquipped;
+                                        this->fishLength = temp1;
+                                        this->isLoach = temp2;
```

Comment:
Assuming attempts have already been made to remove the temp with chain assignments? They have already been used in this actor before now, right?